Assets/Scripts/PlayerController/PlayerMovement.cs

```diff
@@ -70,7 +71,13 @@ void Update()
 
         float t = Time.deltaTime;
         float t2 = t * t;
-        m_CharacterController.Move(new Vector3(m_Velocity.x, m_Velocity.y * t + 0.5f * GetGravityMagnitude() * t2, m_Velocity.z));
+
+        if (m_cannotMove)
+        {
+            m_CharacterController.Move(new Vector3(0, m_Velocity.y * t + 0.5f * GetGravityMagnitude() * t2, 0));
+        } else {
+            m_CharacterController.Move(new Vector3(m_Velocity.x, m_Velocity.y * t + 0.5f * GetGravityMagnitude() * t2, m_Velocity.z));
+        }
```

Comment:
Perhaps you should make this check inside `HandleMovement` instead? You could just do 

```c#
if (m_cannotMove)
{
    m_Velocity = new Vector3(0, m_Velocity.y, 0);
    return;
}
```
which might be a more appropriate abstraction